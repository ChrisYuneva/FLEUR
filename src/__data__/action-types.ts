export const GOODS = {
    REQUEST: 'GOODS_REQUEST',
    SUCCESS: 'GOODS_SUCCESS',
    FAILURE: 'GOODS_FAILURE',
    SET_GENDER: 'GOODS_SET_GENDER',
    SET_TYPE: 'GOODS_SET_TYPE',
    // SET_TITLE: 'GOODS_SET_TITLE'
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};
