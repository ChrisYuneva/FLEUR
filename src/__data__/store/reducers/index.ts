import { combineReducers } from 'redux';
import goodsReducer, { GoodsState } from './goods';
import basketReducer, { BasketState } from './basket';

export type AppStore = {
    goods: GoodsState;
    basket: BasketState;
};

export default combineReducers<AppStore>({
    goods: goodsReducer,
    basket: basketReducer,
});
