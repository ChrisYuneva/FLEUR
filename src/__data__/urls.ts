import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('repos');

export const baseUrl = navigations['repos'];
export const URLs = {
    basket: {
        url: navigations['link.repos.basket'],
    },
    home: {
        url: navigations['link.repos.home'],
    },
    showcase_dress: {
        url: navigations['link.repos.showcase_dress'],
    },
    showcase_tshirts_women: {
        url: navigations['link.repos.showcase_tshirts_women'],
    },
    showcase_jackets_women: {
        url: navigations['link.repos.showcase_jackets_women'],
    },
    showcase_shirts_women: {
        url: navigations['link.repos.showcase_shirts_women'],
    },
    showcase_hoodies_women: {
        url: navigations['link.repos.showcase_hoodies_women'],
    },
    showcase_jeans_women: {
        url: navigations['link.repos.showcase_jeans_women'],
    },
    showcase_skirts: {
        url: navigations['link.repos.showcase_skirts'],
    },
    showcase_accessories: {
        url: navigations['link.repos.showcase_accessories'],
    },
    home_man: {
        url: navigations['link.repos.home_man'],
    },
};
