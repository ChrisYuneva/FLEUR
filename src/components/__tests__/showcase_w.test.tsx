import React from 'react';
import { mount } from 'enzyme';
import { ShowcaseWoman } from '../index';
import { describe, it, expect } from '@jest/globals';
import { Provider } from 'react-redux';
import { store } from '../../__data__/store';

describe('<ShowcaseWoman />', () => {
    const title = 'title';
    const caption = 'caption';

    it('Отрисовывается без ошибок', () => {
        const wrapper = mount(
            <Provider store={store}>
                <ShowcaseWoman title={title} caption={caption} />
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
