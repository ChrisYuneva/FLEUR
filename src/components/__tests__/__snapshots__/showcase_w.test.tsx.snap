// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowcaseWoman /> Отрисовывается без ошибок 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowcaseWoman)
    caption="caption"
    title="title"
  >
    <ShowcaseWoman
      caption="caption"
      setGender={[Function]}
      setType={[Function]}
      title="title"
    >
      <div
        className="wrapper"
      >
        <nav
          className="nav"
        >
          <Link
            className="name"
            to="/home"
          >
            <a
              className="name"
              href="/home"
            >
              FLEUR
            </a>
          </Link>
          <button
            className="link"
            onClick={[Function]}
          >
            Платья
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Куртки
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Рубашки
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Толстовки
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Футболки
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Джинсы
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Юбки
          </button>
          <button
            className="link"
            onClick={[Function]}
          >
            Аксессуары
          </button>
          <Link
            className="link"
            onClick={[Function]}
            to="/showcase_men"
          >
            <a
              className="link"
              href="/showcase_men"
              onClick={[Function]}
            >
              Мужчины
            </a>
          </Link>
        </nav>
        <div
          className="div"
        >
          <header
            className="header"
          >
            <Link
              className="ref"
              to="/basket"
            >
              <a
                className="ref"
                href="/basket"
              >
                <img
                  className="box"
                  src="FILE_MOCK"
                />
                <a
                  className="text"
                >
                  Корзина
                </a>
              </a>
            </Link>
          </header>
          <div
            className="scan"
          >
            <h2>
              title
            </h2>
            <span
              className="span"
            >
              caption
            </span>
            <div
              className="show"
            >
              <div
                className="showrow"
              >
                <Connect(Showcases)
                  initGender="FEMALE"
                  initType="dress"
                >
                  <Showcases
                    addProducts={[Function]}
                    gender={null}
                    getProducts={[Function]}
                    goodsItems={Array []}
                    initGender="FEMALE"
                    initType="dress"
                    loading={true}
                    type={null}
                  >
                    <p>
                      Пожалуйста, подождите. Идёт загрузка
                    </p>
                  </Showcases>
                </Connect(Showcases)>
              </div>
            </div>
          </div>
        </div>
      </div>
    </ShowcaseWoman>
  </Connect(ShowcaseWoman)>
</Provider>
`;
